eip-0020.md

```diff
@@ -0,0 +1,202 @@
+After parsing a payment request the wallet can build a new transaction, sign it
+using a secret key stored locally on the device and then send to the blockchain.
+
+However, in general, this only can be done for _simple_ transactions like
+transferring ERGs and assets between Pay-To-Public-Key addresses or transactions
+which only spend boxes from P2PK addresses.
+
+The reason for this is simple. In Ergo's eUTXO model a box can be protected by
+an arbitrarily complex contract (aka spending condition) and any spending
+transaction should satisfy that condition by adding required context variables,
+creating expected number of outputs with specific registers depending on the
+contract. There is no way for a universal wallet application to know all
+the specific details of all the possible contracts.
+
+On the contrary, every Ergo dApp (think of website) is usually built on top of
+specific contracts. Business logic of the dApp includes spending of the boxes
+protected by those application-specific contracts.
+In other words, any dApp is in full control of creating transactions and thus
+can spend its own contracts without above mentioned problems. However, the dApp
+cannot sign those transactions, because signing requires knowledge of private keys
+which are not stored on the dApp, but instead stored on the mobile wallet.
+
+Thus, interaction between a dApp and a wallet is required such that:
+1) the dApp builds a transaction and makes it available for the wallet application
+2) wallet app shows a confirmation screen to the user, displaying the inboxes and 
+outboxes it is going to sign
+3) when user confirms, wallet app signs the transaction and submits it to the blockchain or
+returns back to the dApp
+4) dApp monitors the transaction on the blockchain and upon confirmations
+proceeds with its business logic.
+
+One possible implementation of such an interaction scheme is described in
+[EIP-0012](https://github.com/ergoplatform/eips/pull/23). It, however, only
+suitable for in-browser wallets, which can interact with dApps via javascript
+code injected into a web context.
+
+Luckily, the design of Ergo contracts allows for a simple and universal
+implementation suitable for mobile or desktop wallets. This EIP describes the interaction
+protocol and the [Reference Implementation](#reference-implementation).
+
+In this 3 step process the main question is: 
+"How the transaction built in dApp can be transferred to the mobile wallet
+application so that it can be signed there?".
+
+## ErgoPay Interaction Protocol
+
+An ErgoPay interaction between Wallet, and dApp is driven by dApp's
+user and proceeds as the following:
+
+1) The user enters the necessary information in the dApp's UI and proceeds to a
+payment screen.
+
+2) The payment screen shows transaction details and a QR code.
```

Comment:
It is needed to show bytes also, then support can be added to desktop wallets as well

Reply:
Either that, another way to go is to use an URI scheme `ergopay:<ReducedTransaction, base 64 url safe encoded>` mentioned earlier in the document. This way, browser can hand over the data to desktop wallets automatically, and it works on mobile browser to mobile wallet as well.